components/core/tools/scripts/lib_install/install-curl.sh

```
@@ -16,8 +16,10 @@ if [ ${EUID:-$(id -u)} -ne 0 ] ; then
   sudo echo "Script can elevate privileges."
 fi
 
-# Get number of cpu cores
-num_cpus=$(grep -c ^processor /proc/cpuinfo)
+# Get number of cpu cores, capped by available memory (min 2 GB per core)
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd)"
+source "${script_dir}/../../../../../tools/scripts/compute-cpp-max-parallelism.sh"
+num_cpus=$compute_cpp_max_parallelism_result
```

Shellcheck (0.11.0)

[warning] compute_cpp_max_parallelism_result is referenced but not assigned.

(SC2154)